From d1610528bf1e9853626957cc7e715f703a10937c Mon Sep 17 00:00:00 2001
From: Ondrej Mular <omular@redhat.com>
Date: Wed, 27 Jan 2016 17:10:24 +0100
Subject: [PATCH] fix updating cluster properties from web UI

Proper handling of checkboxes sent from older version of GUI.
---
 pcsd/remote.rb | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/pcsd/remote.rb b/pcsd/remote.rb
index d166fde..9805baf 100644
--- a/pcsd/remote.rb
+++ b/pcsd/remote.rb
@@ -1910,6 +1910,19 @@ def update_cluster_settings(params, request, session)
   properties = params['config']
   to_update = []
   current = getAllSettings(session)
+
+  # We need to be able to set cluster properties also from older version GUI.
+  # This code handles proper processing of checkboxes.
+  # === backward compatibility layer start ===
+  params['hidden'].each { |prop, val|
+    next if prop == 'hidden_input'
+    unless properties.include?(prop)
+      properties[prop] = val
+      to_update << prop
+    end
+  }
+  # === backward compatibility layer end ===
+
   properties.each { |prop, val|
     val.strip!
     if not current.include?(prop) and val != '' # add
-- 
1.8.3.1

